src/resolvers/index.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+const Multiaddr = require('..') // eslint-disable-line  no-unused-vars
+
+/**
+ * Resolver for dnsaddr addresses.
+ *
+ * @param {Multiaddr} addr
+ * @returns {Promise<Array<string>>}
+ */
+async function dnsaddrResolver (addr) {
+  const { Resolver } = require('./dns')
+  const resolver = new Resolver()
+
+  const peerId = addr.getPeerId()
+  const hostname = addr.toString().split('dnsaddr')[1].split('/')[1]
```

Comment:
`stringTuples()` might be a cleaner method to work with here as you won't need to deal with the /'s.

```js
const [_, hostname] = addr.stringTuples().find(([proto]) => proto === 56 ? true : false) || []
```

You could reverse lookup the code via the codec table to avoid using 56.

This change will also avoid throwing when the address isn't a `dnsaddr`, hostname will just be undefined. Right now the second split will throw.